lib/src/rules/lines_longer_than_80_chars.dart

```diff
@@ -0,0 +1,170 @@
+            name: 'lines_longer_than_80_chars',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(NodeLintRegistry registry) {
+    final visitor = new _Visitor(this);
+    registry.addCompilationUnit(this, visitor);
+  }
+}
+
+class _Visitor extends SimpleAstVisitor {
+  final LintRule rule;
+
+  _Visitor(this.rule);
+
+  @override
+  void visitCompilationUnit(CompilationUnit node) {
+    final lineInfo = node.lineInfo;
+
+    final allowedLineVisitor = new _AllowedLongLineVisitor(lineInfo);
+    node.accept(allowedLineVisitor);
+    final allowedCommentVisitor = new _AllowedCommentVisitor(lineInfo);
+    node.accept(allowedCommentVisitor);
+
+    final allowedLines = []
+      ..addAll(allowedLineVisitor.allowedLines)
+      ..addAll(allowedCommentVisitor.allowedLines);
+
+    final lineCount = lineInfo.lineCount;
+    for (int i = 0; i < lineCount; i++) {
+      final start = lineInfo.getOffsetOfLine(i);
+      final end = i == lineCount - 1
+          ? node.end
+          : lineInfo.getOffsetOfLineAfter(start) - 1;
+      final length = end - start;
+      if (length > 80 && !allowedLines.contains(i + 1)) {
+        rule.reporter.reportErrorForOffset(rule.lintCode, start, length);
+      }
+    }
+  }
+}
+
+class _AllowedLongLineVisitor extends RecursiveAstVisitor {
+  _AllowedLongLineVisitor(this.lineInfo);
+
+  final LineInfo lineInfo;
+  final allowedLines = <int>[];
+
+  @override
+  visitStringInterpolation(StringInterpolation node) {
+    if (node.isMultiline) {
+      _handleMultilines(node);
+    } else {
+      final value = node.elements.map((e) {
+        if (e is InterpolationString) return e.value;
+        if (e is InterpolationExpression) return ' ' * e.length;
+      }).join();
+      _handleSingleLine(node, value);
+    }
+  }
+
+  @override
+  visitSimpleStringLiteral(SimpleStringLiteral node) {
+    if (node.isMultiline)
+      _handleMultilines(node);
+    else
+      _handleSingleLine(node, node.value);
+  }
+
+  _handleMultilines(SingleStringLiteral node) {
+    final startLine = lineInfo.getLocation(node.offset).lineNumber;
+    final endLine = lineInfo.getLocation(node.end).lineNumber;
+    for (var i = startLine; i <= endLine; i++) {
+      allowedLines.add(i);
+    }
+  }
+
+  _handleSingleLine(AstNode node, String value) {
+    if (_looksLikeUriOrPath(value)) {
+      final line = lineInfo.getLocation(node.offset).lineNumber;
+      allowedLines.add(line);
+    }
+  }
+}
+
+class _AllowedCommentVisitor extends GeneralizingAstVisitor {
+  _AllowedCommentVisitor(this.lineInfo);
+
+  final LineInfo lineInfo;
+  final allowedLines = <int>[];
+
+  @override
+  visitNode(AstNode node) {
+    final comments = <CommentToken>[]
+      ..addAll(_getPrecedingComments(node.beginToken))
+      ..addAll(_getPrecedingComments(node.endToken));
+    for (final comment in comments) {
+      final content = comment.toString();
```

Comment:
If the content is less than 80 characters, then I think you don't need to add the line containing it to the list of lines that are allowed to be long, which would save some string manipulations. (See also line 119.)

Reply:
I guess I was assuming that the preferred style would be to move such a comment to the line before, which would mean that we _did_ want to flag the line (because it could be "fixed"). For example:
```
if (condition) {
  return some.really.long.expression; // some really long comment describing why we're returning
}
```
would become
```
if (condition) {
  // some really long comment describing why we're returning
  return some.really.long.expression;
}
```
Come to think of it, do we really want to exempt comment lines? Should we flag a line like
```
// some really long comment that someone put on a single line and didn't bother to break up.
```
It's perfectly reasonable for such a comment to be broken before column 80 and wrap to the next line:
```
// some really long comment that someone put on a single line and didn't bother
// to break up.
```